src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -808,10 +808,11 @@ namespace smt::noodler {
         bool contains_word_equations = !this->m_word_eq_todo_rel.empty();
         bool contains_word_disequations = !this->m_word_diseq_todo_rel.empty();
         bool contains_conversions = !this->m_conversion_todo.empty();
+        bool contains_eqs_and_diseqs_only = this->m_not_contains_todo_rel.empty() && this->m_conversion_todo.empty();
 
-        // As a heuristic, for the case we have exactly one constraint, which is of type 'x (not)in RE', we use universality/emptiness
-        // checking of the regex (using some heuristics) instead of constructing the automaton of RE. The construction (especially complement)
-        // can sometimes blow up, so the check should be faster.
+        // As a heuristic, for the case we have exactly one constraint, which is of type 'x notin RE', we use universality
+        // checking instead of constructing the automaton for complement of RE. The complement can sometimes blow up, so
+        // universality checking should be faster.
```

Comment:
This seems like rebase mistake, can you return it to what is in the devel?

Reply:
```suggestion
        // As a heuristic, for the case we have exactly one constraint, which is of type 'x (not)in RE', we use universality/emptiness
        // checking of the regex (using some heuristics) instead of constructing the automaton of RE. The construction (especially complement)
        // can sometimes blow up, so the check should be faster.
```